src/nupic/math/ArrayAlgo.hpp

```diff
@@ -33,7 +33,7 @@
 #include <iterator>
```

Comment:
There is one bug fix in this .hpp with its own comment below. The rest of the changes in this file are the result of the editor eliminating trailing whitespace.